cli/src/semgrep/commands/ci.py

```diff
@@ -298,6 +299,8 @@ def ci(
                 # so that metadata of current commit is correct
                 if scan_handler:
                     try:
+                        proj_config = ProjectConfig.load_all()
+                        metadata_dict = {**metadata_dict, **proj_config.to_dict()}
```

Comment:
So it looks like semgrepconfig keys will take precedence over metadata set elsewhere - is that a deliberate choice/something we need to document?

Reply:
Yes this is intentional since if metadata/tags are checked into an SCM, then it should be the source of truth. I viewed this as a similar process to terraform: once a resource is managed by terraform, it will overwrite changes made, for example, in the console.